source/simulator/helper_functions.cc

```diff
@@ -2271,17 +2271,53 @@ namespace aspect
 
   template <int dim>
   void
-  Simulator<dim>::replace_outflow_boundary_ids(const unsigned int offset)
+  Simulator<dim>::replace_outflow_boundary_ids(const unsigned int offset,
+                                               const bool is_composition,
+                                               const unsigned int composition_index)
   {
     const Quadrature<dim-1> &quadrature_formula = introspection.face_quadratures.temperature;
 
+    bool consider_darcy_velocity = false;
+    bool consider_melt_velocity = false;
+    unsigned int porosity_idx = numbers::invalid_unsigned_int;
+    if (is_composition)
+      {
+        if (introspection.compositional_field_methods[composition_index] == Parameters<dim>::AdvectionFieldMethod::fem_darcy_field)
+          {
+            consider_darcy_velocity = true;
+            porosity_idx = introspection.find_composition_type(CompositionalFieldDescription::porosity);
+          }
+        if (introspection.compositional_field_methods[composition_index] == Parameters<dim>::AdvectionFieldMethod::fem_melt_field)
+          consider_melt_velocity = true;
```

Comment:
The porosity itself is not an `fem_melt_field` (see for example test advect_field_with_melt_velocity.prm, which is the only one I have found that actually uses the melt fields), so `consider_melt_velocity` should additionally be true if both the model includes melt transport and the composition with index `composition_index` is a field of type porosity.